addon-api/content-script/Tab.js

```diff
@@ -27,4 +40,16 @@ export default class Tab {
       return Promise.resolve();
     }
   }
+  /**
+   * @type {string} editor mode (or empty string for non-editors).
+   */
+  get editorMode() {
+    const pathname = location.pathname.toLowerCase();
+    const split = pathname.split("/").filter(Boolean);
+    if (!split[0] || split[0] !== "projects") return "";
+    if (split.includes("editor")) return "editor";
+    if (split.includes("fullscreen")) return "fullscreen";
+    if (split.includes("embed")) return "embed";
```

Comment:
Note that Scratch Addons won't run on iframes! But users can still go to the embed URL in a new tab, and Scratch Addons will run there.